packages/smart-contracts/src/contracts/ERC20SingleRequestProxy.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import './interfaces/ERC20FeeProxy.sol';
+
+/**
+ * @title ERC20SingleRequestProxy
+ * @notice This contract is used to send a single request to a payee with a fee sent to a third address for ERC20
+ */
+
+contract ERC20SingleRequestProxy {
+  address public payee;
+  address public tokenAddress;
+  address public feeAddress;
+  uint256 public feeAmount;
```

Comment:
Why do we specify the `feeAmount` to be paid, but not the `paymentAmount` ?

Reply:
The contracts always transfer the full balance they have to the `feeProxies.` This would allow partial payments and ensure that the contracts never hold any funds that they should not have.

So instead of hardcoding the `paymentAmount` we use the contract balance.